src/VisualStudio/Core/Def/Implementation/Remote/RemoteHostClientServiceFactory.SolutionChecksumUpdater.cs

```diff
@@ -122,9 +123,10 @@ private void EnqueueChecksumUpdate()
                     var solution = _service.Workspace.CurrentSolution;
                     var checksum = await solution.State.GetChecksumAsync(cancellationToken).ConfigureAwait(false);
 
-                    await remoteHostClient.RunOnRemoteHostAsync(
-                        WellKnownRemoteHostServices.RemoteHostService, solution,
-                        nameof(IRemoteHostService.SynchronizePrimaryWorkspaceAsync), checksum, cancellationToken).ConfigureAwait(false);
+                    await remoteHostClient.IgnoreCancellationAsync(() =>
+                        remoteHostClient.RunOnRemoteHostAsync(
+                            WellKnownRemoteHostServices.RemoteHostService, solution,
+                            nameof(IRemoteHostService.SynchronizePrimaryWorkspaceAsync), checksum, cancellationToken)).ConfigureAwait(false);
```

Comment:
❓ This one is *not* intuitive to me. The caller has the ability to specify a cancellation token, yet we're hiding that cancellation from the resulting task. Is there any other way to implement this check at a higher level?

Reply:
answered in different PR - https://github.com/dotnet/roslyn/pull/22968#discussion_r148895378